beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -2949,17 +2950,14 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         ops.push(StoreOp::PutBlock(block_root, signed_block.clone()));
         ops.push(StoreOp::PutState(block.state_root(), &state));
 
-        if let Some(blobs) = blobs? {
-            if blobs.blobs.len() > 0 {
-                //FIXME(sean) using this for debugging for now
-                info!(self.log, "Writing blobs to store"; "block_root" => ?block_root);
-                ops.push(StoreOp::PutBlobs(block_root, blobs));
```

Comment:
I see that the `ops.push(StoreOp::PutBlobs(block_root, blobs))` method has been removed here and replaced with `self.store.put_blobs(&block_root, (&*blobs).clone())` on line 3004.

I also notice you've put some effort into the writing the [`do_atomically_with_block_and_blobs_cache`](https://github.com/sigp/lighthouse/blob/5eee07c33b87be7b8a9761d12fb1abc9d8b8ad26/beacon_node/store/src/hot_cold_store.rs#L825-L899) function, but I don't see where`StoreOp::PutBlobs` is constructed anymore.

I would expect this block of code (where I'm applying this comment) remains and the newly-added block containing line 3004 would be removed.

I could be missing something, though!

Reply:
Originally, blobs always went into the `hot_db` here, along with blocks, execution payload, etc. Since that line of code is not in a loop we are only putting one blobs sidecar in store. I knew we wanted to cache blobs so I used `put_blobs`. However, I wasn't sure if I we wanted to put blocks in cache here too, but here I got the answer https://github.com/sigp/lighthouse/pull/3931 and will use `do_atomically_with_block_and_blobs_cache`.